### PR TITLE
PSBT: add no_publish flag for safe batch channel opens

### PR DESCRIPTION
**Pull request by guggero**
To make it safer to open multiple channels in one batch, we add the `--no_publish` flag to `lncli openchannel` (called `no_publish` in the `PsbtShim` in RPC).

When opening multiple channels in the same transaction, the `--no_publish` flag should be set for every channel but the last one. That way only after the last channel is negotiated successfully the batch transaction is published to the network. If that isn't done, it could lead to loss of funds if one of the peers involved times out and no commitment transactions were signed for that channel.

### CODE REVIEW

**Comment by alexbosworth** [comment]
What if you are doing a batch open across multiple nodes?

**Comment by guggero** [comment]
Then you would only set the flag to `false` on one of the nodes for exactly one channel.

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -1653,6 +1653,16 @@ message PsbtShim {
...
+
+    /*
+    If a channel should be part of a batch (multiple channel openings in one
+    transaction), it can be dangerous if the whole batch transaction is
+    published too early before all channel opening negotiations are completed.
+    This flag prevents this particular channel from broadcasting the transaction
+    after the negotiation with the remote peer. In a batch of channel openings
+    this flag should be set to true for every channel but the very last.
+    */
+    bool no_publish = 3;
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply lnrpc/rpc.proto:1665]: Hmm, yeah that seems like a good choice, other than the breaking behavior....

Given that this is out in the wild, we should definitely do _something_ to make it a bit safer. I think also the docs included in this PR go a long ways, as they actually explain some the risks involved in using the API incorrectly.

File: cmd/lncli/cmd_open_channel.go [hunk 2]
```diff
@@ -163,6 +163,15 @@ var openChannelCommand = cli.Command{
 				"as a base and add the new channel output to " +
 				"it instead of creating a new, empty one.",
 		},
+		cli.BoolFlag{
+			Name: "no_publish",
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply cmd/lncli/cmd_open_channel.go:167]: So in this case `lnd` doesn't really know that this is a batch funding transaction, only that it received a partial PSBT. For the "official" batch funding implementation, i think this is something we'll want to do.

**Comment by cfromknecht** [comment]
Taking over in #4455

**Pull request closed without merging**